duration period
uint64 iterations
---
bool success
